### ISSUE TITLE
automate mail creation in fastmail

### ISSUE THREAD

**Issue by benoitc** [erlef/infra-wg#10]
As suggested by Bryan we can automate the creation of mail acount from the wildapricot api once #8

**Comment by benoitc** [comment]
some questions in vrac:
* Where would be hosted that script and the credentials for it? 
* How could we monitor it?

**Comment by starbelly** [comment]
@benoitc so it would just be a simple script, we could keep it in this repo perhaps, there would be nothing sensitive in it, keys would be pulled in via env vars. 

Let's define what it would do: 

## Prerequisite : 
Presumably members could login to their wildparicot account and tick a box or two stating they would like an erlef address, whether it should be an alias/fowarding address, etc. There would also have to be a field which only an administrator (or a script that has access) could populate that indictates the member has an active address. 

## Mostly Manual method (essentially)

 - We can batch query the wildapricot for the attributes listed above (or literally iterate over all members)
 - This could output a a csv of member accounts for which a user would like an erlef address and do not currently have one. 
 - The administrator could then manually go through add an address, notify the member of their address, temporary password, etc. 


## Programmatic 

  All the above, except we create the email accounts automagically. 

  - Query wild apricot
  - Login to our provider programmatically, any language will do, hold on to the cookie for the duration of  script 
  - Interact with the their jmapi (just http) to create an account 
  - Send member an email
  
At with any thing like this, I would run this quite a few times before putting it on a cronjob. To that, I would a dry run option :)

**Comment by starbelly** [comment]
I'm going to take another stab at this after I get a few other things done. The problem to be clear was getting logged in their main api to perform administrative tasks.

**Comment by benoitc** [comment]
what's the status?

**Comment by starbelly** [comment]
I never looked back into this, it isn't worth the time right now. If someone wants to take a stab at it, by all means feel free. Let me add some color here:

- fastmail api for this (or in general) is not documented
- you can login as a normal user by following a few articles or reverse engineering what happens in fastmail UI
- What I could not do was login as an admin. A few hours got sunk into this and eventually pulled the plug in favor of manually processing, and considering the volume of email requests (or lack thereof) proved to be a good decision.

**Comment by benoitc** [comment]
i think before going to automate it, we should document what is actually done for it.  So let's close it for now and focus on #50?

**Comment by starbelly** [comment]
@benoitc yes, and I started on that (in the wiki). The doc could be better though. Will carve out more time this week to take more stabs. 

I see you're asking for status : https://github.com/erlef/infra-wg/wiki
